mayastor/src/lvs/lvs_pool.rs

```diff
@@ -309,15 +349,35 @@ impl Lvs {
                 } => Ok(parsed.get_name()),
                 _ => Err(Error::InvalidBdev {
                     source: e,
-                    msg: "base_bdev for the pool does not match".into(),
+                    name: args.disks[0].clone(),
                 }),
             },
             Ok(name) => Ok(name),
         }?;
 
         match Self::import(&args.name, &bdev).await {
             Ok(pool) => Ok(pool),
-            Err(_) => Self::create(&args.name, &bdev).await,
+            Err(Error::Import {
+                source,
+                name,
+            }) if source == Errno::EINVAL => {
+                // there is a pool here, but it does not match the name
+                error!("pool name mismatch");
+                Err(Error::Import {
+                    source,
+                    name,
+                })
+            }
+            // else some other error, try to create the the pool
+            Err(Error::Import {
+                source, ..
+            }) if source == Errno::EILSEQ => {
+                Self::create(&args.name, &bdev).await
```

Comment:
if the import or the create fails should we destroy the base bdev (though only for the case where we've just created it) ?

Reply:
Yes good idea, I've added the deletion it and error!() in the logs. If we fail to delete, not sure what else we can do...